CONTRIBUTING.md

```diff
@@ -0,0 +1,147 @@
+# Raiden Development Guide
+
+Welcome! This guide serves as the guideline to contributing to the Raiden Network
+codebase. It's here to help you understand what developmenr practises we use here
+and what are the requirements for a Pull Request to be opened against Raiden.
+
+- [Contributing](#contributing)
+    - [Creating an Issue](#creating-an-issue)
+    - [Creating a Pull Request](#creating-a-pull-request)
+- [Development Guidelines](#development-guidelines)
+    - [Coding Style](#coding-style)
+    - [Workflow](#workflow)
+
+
+## Contributing
+
+You can contribute to the development with two basic methods. You can either open
+an Issue or if you have programming abilities open a Pull Request.
+
+### Creating an Issue
+
+If you experience a problem while using Raiden or want to request a feature then you should open an issue against the repository. All issues should contain:
+
+**For Feature Requests:**
+- A description of what you would like to see implemented
+- An explanation of why you believe this would make a good addition to Raiden
+
+**For Bugs:**
+- A short description of the problem
+- Detailed description of your system, raiden version, geth version, solidity version e.t.c.
+- What was the exact unexpected thing that occured
+- What you were expecting to happen instead
+
+### Creating a Pull Request
+
+If you have some coding abilities and would like to contribute to the actual codebase of Raiden then you can open a Pull Request(PR) against the repository.
+
+All PRs should be:
+- Self-contained
+- As short as possible and address a single issue or even a part of an issue.
+If the PR adds a new feature and it's very long, consider breaking into smaller ones.
```

Comment:
Could be worded as something like `consider breaking very long PRs into smaller ones` independent of being a new feature or not.